src/commands/lambda/constants.ts

```diff
@@ -1,46 +1,49 @@
 export const DD_LAMBDA_EXTENSION_LAYER_NAME = 'Datadog-Extension'
 export const EXTENSION_LAYER_KEY = 'extension'
-export const RUNTIME_LAYER_LOOKUP = {
+export const DOTNET_RUNTIME = 'dotnetcore3.1'
+export const LAYER_LOOKUP = {
   [EXTENSION_LAYER_KEY]: DD_LAMBDA_EXTENSION_LAYER_NAME,
+  'dotnetcore3.1': 'dd-trace-dotnet',
   'nodejs12.x': 'Datadog-Node12-x',
   'nodejs14.x': 'Datadog-Node14-x',
   'python3.6': 'Datadog-Python36',
   'python3.7': 'Datadog-Python37',
   'python3.8': 'Datadog-Python38',
   'python3.9': 'Datadog-Python39',
 } as const
-// We exclude the Extension Layer Key in order for the runtime
-// to be used directly in HANDLER_LOCATION.
-export type Runtime = Exclude<keyof typeof RUNTIME_LAYER_LOOKUP, typeof EXTENSION_LAYER_KEY>
-
-export const ARM_RUNTIMES = [EXTENSION_LAYER_KEY, 'python3.8', 'python3.9']
-export const ARM64_ARCHITECTURE = 'arm64'
-export const ARM_LAYER_SUFFIX = '-ARM'
 
 export enum RuntimeType {
+  DOTNET,
+  CUSTOM,
+  JAVA,
   NODE,
   PYTHON,
+  RUBY,
 }
 
-export const RUNTIME_LOOKUP: {[key: string]: RuntimeType} = {
+export const RUNTIME_LOOKUP = {
+  'dotnetcore3.1': RuntimeType.DOTNET,
+  java11: RuntimeType.JAVA,
+  'java8.al2': RuntimeType.JAVA,
   'nodejs12.x': RuntimeType.NODE,
   'nodejs14.x': RuntimeType.NODE,
+  'provided.al2': RuntimeType.CUSTOM,
   'python3.6': RuntimeType.PYTHON,
   'python3.7': RuntimeType.PYTHON,
   'python3.8': RuntimeType.PYTHON,
   'python3.9': RuntimeType.PYTHON,
+  'ruby2.5': RuntimeType.RUBY,
+  'ruby2.7': RuntimeType.RUBY,
 }
 
-const PYTHON_HANDLER_LOCATION = 'datadog_lambda.handler.handler'
-const NODE_HANDLER_LOCATION = '/opt/nodejs/node_modules/datadog-lambda-js/handler.handler'
-export const HANDLER_LOCATION = {
-  'nodejs12.x': NODE_HANDLER_LOCATION,
-  'nodejs14.x': NODE_HANDLER_LOCATION,
-  'python3.6': PYTHON_HANDLER_LOCATION,
-  'python3.7': PYTHON_HANDLER_LOCATION,
-  'python3.8': PYTHON_HANDLER_LOCATION,
-  'python3.9': PYTHON_HANDLER_LOCATION,
-}
+export type Runtime = keyof typeof RUNTIME_LOOKUP
+export type LayerName = keyof typeof LAYER_LOOKUP
```

Comment:
I would choose a different name here.

Right now `LayerName` is defined as the **keys** of `LAYER_LOOKUP` (`dotnetcore3.1`, `nodejs12.x`, etc.). 

However, it looks like the layer names are actually the **values** of `LAYER_LOOKUP` (`dd-trace-dotnet`, `Datadog-Node12-x`, etc.).

I think I would name this `LayerKeys`. That reflects that they are the keys that let us get the layer name, not the layer names themselves.